### ISSUE TITLE
[BUG] `viash run` does not pick rebuild images when changes are made to the docker platform.

### ISSUE THREAD

**Issue by DriesSchaumont** [viash-io/viash#656]
### What happened?

`viash run` does not properly detect changes to the docker platform and does not build a new docker image when needed.

### Steps to reproduce

Make sure no docker image is present to start with
```
docker rmi $(docker images -q 'dummycomponent/myscript' | uniq)
```

```
mkdir -p /tmp/test_bug
cd /tmp/test_bug
cat > _viash.yaml << 'HERE'
viash_version: 0.8.5

source: src
target: target
HERE

mkdir -p src/dummycomponent

cat > src/dummycomponent/config.vsh.yaml << 'HERE'
functionality:
  name: myscript
  namespace: "dummycomponent"
  resources:
    - type: python_script
      path: script.py
      text: |
        print("OK!", flush=True)
platforms:
  - type: docker
    image: python:latest
  - type: nextflow
HERE

viash run ./src/dummycomponent/config.vsh.yaml


cat > src/dummycomponent/config.vsh.yaml << 'HERE'
functionality:
  name: myscript
  namespace: "dummycomponent"
  resources:
    - type: python_script
      path: script.py
      text: |
        import pandas
        print("OK!", flush=True)
platforms:
  - type: docker
    image: python:latest
    setup:
      - type: python
        packages: 
          - pandas
  - type: nextflow
HERE

viash run ./src/dummycomponent/config.vsh.yaml
```

### Expected behavior

A change to the docker platform should be reflected in the docker image by default.
During the second run, the `pandas` package was added to the config, but the package is missing.
After explicitly building the container with `-- ---setup build`, the container image is forcefully build and `viash run` works.

### Relevant log output

```
viash run ./src/dummycomponent/config.vsh.yaml
[notice] Checking if Docker image is available at 'dummycomponent/myscript:latest'
[warning] Could not pull from 'dummycomponent/myscript:latest'. Docker image doesn't exist or is not accessible.
[notice] Building container 'dummycomponent/myscript:latest' with Dockerfile
OK!
Traceback (most recent call last):
  File "/viash_automount/tmp//viash-run-myscript-ACExuP.py", line 25, in <module>
    import pandas
ModuleNotFoundError: No module named 'pandas'
Files and logs are stored at '/tmp/viash_myscript11921559591400468098'
```

However after building the container with `-- ---setup build`

```
$ viash run ./src/dummycomponent/config.vsh.yaml -- ---setup build
$ viash run ./src/dummycomponent/config.vsh.yaml
OK!
```

### Version

- OS: Ubuntu 22.04
- Viash 0.8.5

### Confirmation

- [X] I have searched the existing issues to make sure this is not a duplicate.
- [X] I have provided clear and concise information about the bug.

**Comment by tverbeiren** [comment]
This behaviour is described [in the documentation](https://viash.io/guide/component/add-dependencies.html#step-3-rebuild-docker-image):

> After adding additional setup entries, it’s important to rerun `---setup cachedbuild` to rebuild the Docker image, as Viash will not rebuild the Docker image when it already exists.

AFAIK the behavior of 1) `viash run ...` and 2) first running `viash build ...` and then running the generated executable is the same: If the image exists locally then don't rebuild it.

**Comment by DriesSchaumont** [comment]
> This behaviour is described [in the documentation](https://viash.io/guide/component/add-dependencies.html#step-3-rebuild-docker-image):
> 
> > After adding additional setup entries, it’s important to rerun `---setup cachedbuild` to rebuild the Docker image, as Viash will not rebuild the Docker image when it already exists.
> 
> AFAIK the behavior of 1) `viash run ...` and 2) first running `viash build ...` and then running the generated executable is the same: If the image exists locally then don't rebuild it.

Just to note: this is different to what has been described in the documentation: https://viash.io/reference/config/platforms/docker/#setup_strategy

`ifneedbepullelsecachedbuild`: If the image does not exist locally, pull the image. If the image does exist, build it with caching enabled.

So perhaps the website needs to be updated?

**Comment by tverbeiren** [comment]
You're right, it should say something like: If the image exists, use it. If the image does not exist, try to pull it. As a last resort build it from the Viash config.

**Comment by DriesSchaumont** [comment]
As an added bonus, adding a `ifneedbepullalwayscachedbuild` would be a nice addition 🤷

**Comment by Grifs** [comment]
Documentation will be updated with Viash 0.9.0
